### PR TITLE
[Feature] Rate Limiting Chatbot interaction with chatbot tokens

### PR DESCRIPTION
**Pull request by d3li0n**
# Description
Adds rate limiting on chatbot interaction

## Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This requires a run of `yarn install`

# How Has This Been Tested?
- [x] Added unit tests
- [x] Manually tested each modified component

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules (TBD: there is a separate PR that needs to be reviewed before this will be merged to main)

### CODE REVIEW

File: packages/app/components/Organization/DevelopmentTab.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { Button, Card, message } from 'antd'
+import { ReactElement } from 'react'
+
+export default function DevelopmentTab({
+  organizationId,
+}: {
+  organizationId: number
+}): ReactElement {
+  const populateTable = async () => {
```
Comment (d3li0n) [inline packages/app/components/Organization/DevelopmentTab.tsx:9]: Temporary component to feed the tables with chat tokens

File: packages/server/src/organization/organization.controller.ts [hunk 2]
```diff
@@ -56,10 +56,57 @@ import { In } from 'typeorm';
...
 import { CourseSettingsModel } from 'course/course_settings.entity';
 import { EmailVerifiedGuard } from 'guards/email-verified.guard';
+import { ChatTokenModel } from 'chatbot/chat-token.entity';
+import { v4 } from 'uuid';
 
 @Controller('organization')
 export class OrganizationController {
   constructor(private organizationService: OrganizationService) {}
+
+  @Post(':oid/populate_chat_token_table')
```
Comment (d3li0n) [inline packages/server/src/organization/organization.controller.ts:66]: Temporary endpoint

File: packages/app/components/Chatbot/Chatbot.tsx [hunk 3]
```diff
@@ -42,8 +42,9 @@ export const ChatbotComponent: React.FC = () => {
   const { cid } = router.query
   const profile = useProfile()
   const [isLoading, setIsLoading] = useState(false)
-  const [interactionId, setInteractionId] = useState<number | null>(null)
+  const [_interactionId, setInteractionId] = useState<number | null>(null)
```
Comment (wskksw) [inline packages/app/components/Chatbot/Chatbot.tsx:45]: what does the underscore before do. Online it says is making var private? Just curious

File: packages/app/components/Chatbot/Chatbot.tsx [hunk 4]
```diff
@@ -267,17 +283,25 @@ export const ChatbotComponent: React.FC = () => {
             value={input}
             onChange={(e) => setInput(e.target.value)}
             placeholder="Ask something..."
-            onPressEnter={handleAsk}
+            onPressEnter={input.trim().length > 0 ? handleAsk : undefined}
```
Comment (wskksw) [inline packages/app/components/Chatbot/Chatbot.tsx:286]: nice we can't ask empty questions now!

**Review by wskksw — APPROVED** [review]
Very well done

File: packages/app/components/Chatbot/Chatbot.tsx (hunk 3 above)
Comment (d3li0n) [reply packages/app/components/Chatbot/Chatbot.tsx:45]: linter will complain that the value is unused, adding `_` marks it as "we are aware that it's not being used"

**Review by AdamFipke — APPROVED** [review]
didn't manually test it but code LGTM

**Pull request merged**